pix-editor/app/controllers/authenticated/whitelisted-urls/list.js

```diff
@@ -0,0 +1,111 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class WhitelistedUrlsController extends Controller {
+  @service router;
+  queryParams = ['url', 'names'];
+  @tracked searchUrl = '';
+  @tracked searchNames = '';
+  @tracked url = '';
+  @tracked names = '';
+
+  get filteredWhitelistedUrls() {
+    return this.model.whitelistedUrls.filter((whitelistedUrl) => {
+      const hasMatchingUrl = whitelistedUrl.url.includes(this.url ?? '');
+      let hasMatchingNames = true;
+      if (this.names) {
+        const urlNames = whitelistedUrl.relatedSkillNames ?? [];
+        const searchedNames = this.names.split(',');
+        hasMatchingNames = searchedNames.some((name) => urlNames.includes(name));
+      }
+      return hasMatchingUrl && hasMatchingNames;
+    });
+  }
+
+  formatCheckType = (checkType) => {
+    if (checkType === 'exact_match') {
+      return 'Strictement égale à';
+    } else {
+      return 'Commence par';
+    }
+  };
+
+  checkTypeColor = (checkType) => {
+    if (checkType === 'exact_match') {
+      return 'primary';
+    } else {
+      return 'secondary';
+    }
+  };
+
+  formatCreationString = (whitelistedUrl) => {
+    const date = new Date(whitelistedUrl.createdAt);
+    const DDMMYYYY = this.formatDateToDDMMYYY(date);
+    const HHMM = this.formatDateToHHMM(date);
+    if (!whitelistedUrl.creatorName) {
+      return `${DDMMYYYY} à ${HHMM}`;
+    }
+    return `${DDMMYYYY} à ${HHMM} par ${whitelistedUrl.creatorName} `;
+  };
+
+  formatUpdateString = (whitelistedUrl) => {
+    const date = new Date(whitelistedUrl.updatedAt);
+    const DDMMYYYY = this.formatDateToDDMMYYY(date);
+    const HHMM = this.formatDateToHHMM(date);
+    if (!whitelistedUrl.latestUpdatorName) {
+      return `${DDMMYYYY} à ${HHMM}`;
+    }
+    return `${DDMMYYYY} à ${HHMM} par ${whitelistedUrl.latestUpdatorName}`;
+  };
+
+  formatDateToDDMMYYY(date) {
+    const formater = new Intl.DateTimeFormat('fr');
+    return formater.format(date);
+  }
+
+  formatDateToHHMM(date) {
+    const formater = new Intl.DateTimeFormat('fr', {
+      hour: '2-digit',
+      minute: '2-digit',
+    });
+    return formater.format(date);
+  }
+
+  @action
+  async copyUrl(whitelistedUrl) {
+    await navigator.clipboard.writeText(whitelistedUrl.url);
+  }
+
+  @action
+  async goToEditWhitelistedUrl(whitelistedUrlId, event) {
+    event.preventDefault();
```

Comment:
C’est perturbant d’avoir de la manipulation d’events UI directement dans le controller.

La tendance est plutôt à faire disparaitre les controllers quand on le peut et à débrancher au plus tôt vers un Component.
Ici le controller est nécessaire pour gérer les query params.

Le mélange de logique UI et de manipulation de données rend le code moins lisible.
Et le mélange de méthode pour gérer la recherche et de méthode qui sont en faite des fonctions helpers de formattage (elles n’utilisent pas le this) rend l’ensemble plus compliqué à comprendre.

Reply:
on a créé un component dédié pour la liste et donc c'est géré là-dedans, je pense que la séparation dont tu parles est maintenant satisfaite